src/main/scala/org/broadinstitute/dsde/firecloud/model/ModelSchema.scala

```diff
@@ -14,35 +13,114 @@ import org.broadinstitute.dsde.firecloud.FireCloudConfig
  * Time: 14:18
  */
 
-object ModelSchema {
+trait ModelSchema {
```

Comment:
The more I think about it, the more I feel like these functions should continue returning `Try`s rather than throwing exceptions. If we expect the possibility that an implementation might need to throw an exception (and we already have one case of that), we should build that into the API. I would guess that it would also minimize the need to change client code.